### DISCUSSION TITLE
Is it possible to add a `xwayland-scale` configuration?

### DISCUSSION THREAD

**Discussion by cexossqu** [YaLTeR/niri#524] [Ideas]
Can we add an `Xwayland-scale` configuration to control whether the xwayland application is scaled? Because the current scale will scale the xwayland application together, which will cause the xwayland application to be very blurry, and it is obviously unreasonable to give up scaling for xwayland application.

**Comment by YaLTeR** [comment]
Well... niri doesn't have built-in Xwayland, so there's nothing to control :smile:

**Reply by cexossqu** [reply]
> Well... niri doesn't have built-in Xwayland, so there's nothing to control 😄

Is there a method to prevent niri from scaling specific Windows, similar to the configuration available in window-rule? I am encountering difficulty resolving the xwayland application blur issue in niri when using fractional scaling.

**Reply by YaLTeR** [reply]
Not at the moment. Also, that would probably be somewhat nontrivial to implement.

Are you using a fractional or an integer scale factor?

**Reply by cexossqu** [reply]
> Not at the moment. Also, that would probably be somewhat nontrivial to implement.
> 
> Are you using a fractional or an integer scale factor?

fractional。As far as I know, both Sway and Hyprland have encountered the same issue. While Hyprland has addressed it, my current approach involves adjusting the font size instead of using zoom.

**Comment by grigio** [comment]
are there new workarounds for this?
